src/main/java/appland/telemetry/AppInsightsClient.java

```diff
@@ -0,0 +1,56 @@
+package appland.telemetry;
+
+import com.intellij.openapi.application.ApplicationInfo;
+import com.intellij.util.io.HttpRequests;
+
+import appland.AppMapPlugin;
+import appland.telemetry.data.Tag;
+import appland.telemetry.data.TelemetryEvent;
+import com.google.gson.GsonBuilder;
+
+import org.jetbrains.annotations.NotNull;
+
+
+/*
+ * Based on:
+ * https://github.com/Azure/azure-sdk-for-java/blob/4a53adf6274ced5af8243983042b5e32bac85bd7/sdk/
+ * monitor/azure-monitor-opentelemetry-exporter/src/main/java/com/azure/monitor/opentelemetry/
+ * exporter/AzureMonitorExporterBuilder.java
+ */
+class AppInsightsClient {
+    private final @NotNull String ingestionEndpoint;
+
+    AppInsightsClient(@NotNull String ingestionEndpoint) {
+        this.ingestionEndpoint = ingestionEndpoint;
+    }
+
+    public void track(@NotNull TelemetryEvent event) {
+        var userId = Identity.getMachineId();
+        var osVersion = System.getProperty("os.version");
+
+        event.property("common.os", System.getProperty("os.name"))
+            .property("common.platformversion", System.getProperty("os.version"))
+            .property("common.jvmversion", System.getProperty("java.version"))
+            .property("common.extversion", AppMapPlugin.getDescriptor().getVersion())
+            .property("common.intellijversion", ApplicationInfo.getInstance().getFullVersion())
+            .property("common.source", "IntelliJ")
+            .tag(Tag.OsVersion, osVersion)
+            .tag(Tag.UserId, userId)
+            .tag(Tag.SessionId, Session.getId());
+
+        try {
+            var url = String.format("%s/v2/track", ingestionEndpoint);
+            var gson = new GsonBuilder().create();
+            var json = gson.toJson(event);
+            System.out.println(json);
+            HttpRequests.post(url, "application/json")
+                .tuner(t -> t.setRequestProperty("Accept", "application/json"))
+                .connect(req -> {
+                    req.write(json);
+                    return req;
+                });
+        } catch (Exception e) {
+            e.printStackTrace();
```

Comment:
Better use `Logger.getInstance(getClass()).warning(e);` or `Logger.getInstance(getClass()).error(e);` to log this